packages/core/src/diff/iterator.rs

```diff
@@ -145,15 +145,15 @@ impl VirtualDom {
             // we need to find the right "foothold" though - we shouldn't use the "append" at all
             if left_offset == 0 {
                 // insert at the beginning of the old list
-                let foothold = &old[old.len() - right_offset];
+                let foothold = &new[old.len() - right_offset];
```

Comment:
I'm not entirely sure about this logic. Should we be removing old here? Has it already been removed?